### DISCUSSION TITLE
Configs "fatigue"

### DISCUSSION THREAD

**Discussion by ganigeorgiev** [yiisoft/app#203] [General]
I'm in the process of evaluating yii3 for a new open source project.

So far, I like a lot of the changes and the "self-contained" structure of the packages, but I think the base app template should be simplified a little (more specifically the configurations).

It took me a while and I had to go through the code to get just an idea why there are so many configuration files and what they are used for (even now I'm not completely sure what actually `application-parameters`, `providers`, `delegates` are needed for).

Although I could understand the benefits of yii-config (reminds me a little of symfony-flex), I personally wouldn't use it for small and medium size apps. And even if I decide to use it I'll have to make sure to add at least a detailed comment to each config file why and what is it used for.

Having so many configurations, scattered across so many files for me leads to a configs "fatigue".

Is it just me that thinks this way? To those who prefer having multiple configs, could you elaborate more on your reasoning?

**Comment by samdark** [comment]
Multiple single-purpose configs are easier to grasp than a single big config for me personally. Docs are a bit lacking now but that would be fixed.
